floris/tools/uncertainty_interface.py

```diff
@@ -1,460 +1,217 @@
-            Defaults to None, in which case default PMFs are calculated
-            using values provided in **unc_options**.
-        fix_yaw_in_relative_frame (bool, optional): When set to True, the
-            relative yaw angle of all turbines is fixed and always has the
-            nominal value (e.g., 0 deg) when evaluating uncertainty in the
-            wind direction. Evaluating  wind direction uncertainty like this
-            will essentially come down to a Gaussian smoothing of FLORIS
-            solutions over the wind directions. This calculation can therefore
-            be really fast, since it does not require additional calculations
-            compared to a non-uncertainty FLORIS evaluation.
-            When fix_yaw_in_relative_frame=False, the yaw angles are fixed in
-            the absolute (compass) reference frame, meaning that for each
-            probablistic wind direction evaluation, our probablistic (relative)
-            yaw angle evaluated goes into the opposite direction. For example,
-            a probablistic wind direction 3 deg above the nominal value means
-            that we evaluate it with a relative yaw angle that is 3 deg below
-            its nominal value. This requires additional computations compared
-            to a non- uncertainty evaluation.
-            Typically, fix_yaw_in_relative_frame=True is used when comparing
-            FLORIS to historical data, in which a single measurement usually
-            represents a 10-minute average, and thus is often a mix of various
-            true wind directions. The inherent assumption then is that the turbine
-            perfectly tracks the wind direction changes within those 10 minutes.
-            Then, fix_yaw_in_relative_frame=False is typically used for robust
-            yaw angle optimization, in which we take into account that the turbine
-            often does not perfectly know the true wind direction, and that a
-            turbine often does not perfectly achieve its desired yaw angle offset.
-            Defaults to fix_yaw_in_relative_frame=False.
+        # Save these inputs
+        self.wd_resolution = wd_resolution
+        self.ws_resolution = ws_resolution
+        self.ti_resolution = ti_resolution
+        self.yaw_resolution = yaw_resolution
+        self.power_setpoint_resolution = power_setpoint_resolution
+        self.wd_std = wd_std
+        self.verbose= verbose
+
+        # If wd_sample_points, default to 1 and 2 std
+        if wd_sample_points is None:
+            wd_sample_points = [-2 * wd_std, -1 * wd_std, 0, wd_std, 2 * wd_std]
+
+        self.wd_sample_points = wd_sample_points
+        self.n_sample_points = len(self.wd_sample_points)
+
+        # Get the weights
+        self.weights = self._get_weights(self.wd_std, self.wd_sample_points)
+
+
+        # Call base init function
+        super().__init__(configuration)  # Call the parent's __init__
+
+    def set(self,
```

Comment:
Should bump self down. Odd that ruff didn't catch that.

Reply:
fixed